movies-manager/src/components/form-select/form-select.component.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { Component, ChangeEvent } from "react";
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faAngleDown } from '@fortawesome/free-solid-svg-icons';
+import "./form-select.component.scss";
+
+interface ISelectFormState {
+    genres: string[],
+    selectedGenres: string[],
+    isOpen: boolean;
+}
+
+interface ISelectFormProps {
+    onApplyGenres: (newGenres: string[]) => void;
+    genres: string[];
+}
+
+const allGenres = [
+    'Adventure',
+    'Horror',
+    'Comedy',
+    'Family',
+    'Drama',
+    'Romance',
+];
+
+const blockName = 'form-select';
+
+export class FormSelect extends Component<ISelectFormProps, ISelectFormState> {
+    constructor(props: ISelectFormProps) {
+        super(props);
+        this.state = {
+            genres: Array.from(new Set([ ...allGenres, ...this.props.genres ])),
+            selectedGenres: this.props.genres,
+            isOpen: false,
+        }
+    }
+
+    toggleSelectDropdown = (event: any): void => {
+	    event.preventDefault();
+	    event.stopPropagation();
+        if (!this.state.isOpen) {
```

Comment:
`if (this.state.isOpen) {
            this.setState({ isOpen: false });
            this.props.onApplyGenres(this.state.selectedGenres);
        } else {
            this.setState({ isOpen: true });
        }`

Reply:
done